sdk/core/azure-core/azure/core/pipeline/policies/_authentication.py

```diff
@@ -91,7 +91,7 @@ def on_request(self, request):
 
         if self._need_new_token:
             self._token = self._credential.get_token(*self._scopes)
-            self._update_headers(request.http_request.headers, self._token.token)
+        self._update_headers(request.http_request.headers, self._token.token)
```

Comment:
```suggestion
        if self._token is not None:
            self._update_headers(request.http_request.headers, self._token.token)
```

Reply:
We had a discussion off line and agreed token will not be None because need_new_token makes sure if the the token is None then a new token will be created
```
if self._need_new_token:
    self._token = self._credential.get_token(*self._scopes)
```
```
def _need_new_token(self):
    return not self._token or self._token.expires_on - time.time() < 300
```